mlflow/llama_index/__init__.py

```diff
@@ -417,11 +469,14 @@ def _load_llama_model(path, flavor_conf):
     if model_code_path := pyfunc_flavor_conf.get(MODEL_CODE_PATH):
         # TODO: The code path saved in the MLModel file is the local absolute path to the code
         # file when it is saved. We should update the relative path in artifact directory.
-        model_code_path = os.path.join(
-            path,
-            os.path.basename(model_code_path),
-        )
-        return _load_model_code_path(model_code_path, flavor_conf)
+        model_code_path = os.path.join(path, os.path.basename(model_code_path))
+
+        model_config = pyfunc_flavor_conf.get(MODEL_CONFIG, flavor_conf.get(MODEL_CONFIG, {}))
```

Comment:
```suggestion
        model_config = pyfunc_flavor_conf.get(MODEL_CONFIG) or flavor_conf.get(MODEL_CONFIG, {})
```

to avoid a unnecessary function call.